compiler/noirc_frontend/src/hir/def_collector/dc_mod.rs

```diff
@@ -485,11 +365,69 @@ impl<'a> ModCollector<'a> {
                 errors.push((error.into(), self.file_id));
             }
 
+            // Add all functions that have a default implementation in the trait
+            let mut unresolved_functions =
+                UnresolvedFunctions { file_id: self.file_id, functions: Vec::new() };
+            for trait_item in &trait_definition.items {
+                // TODO(Maddiaa): Investigate trait implementations with attributes see: https://github.com/noir-lang/noir/issues/2629
+                if let TraitItem::Function {
+                    name,
+                    generics,
+                    parameters,
+                    return_type,
+                    where_clause,
+                    body: Some(body),
+                } = trait_item
+                {
+                    let func_id = context.def_interner.push_empty_fn();
+
+                    let impl_method = NoirFunction::normal(FunctionDefinition::normal(
+                        name,
+                        generics,
+                        parameters,
+                        body,
+                        where_clause,
+                        return_type,
+                    ));
+                    unresolved_functions.push_fn(self.module_id, func_id, impl_method);
+                }
+            }
+
+            // Add all functions that have a default implementation in the trait
+            let mut unresolved_functions =
+                UnresolvedFunctions { file_id: self.file_id, functions: Vec::new() };
+            for trait_item in &trait_definition.items {
+                // TODO(Maddiaa): Investigate trait implementations with attributes see: https://github.com/noir-lang/noir/issues/2629
+                if let TraitItem::Function {
+                    name,
+                    generics,
+                    parameters,
+                    return_type,
+                    where_clause,
+                    body: Some(body),
+                } = trait_item
+                {
+                    let func_id = context.def_interner.push_empty_fn();
+
+                    let impl_method = NoirFunction::normal(FunctionDefinition::normal(
+                        name,
+                        generics,
+                        parameters,
+                        body,
+                        where_clause,
+                        return_type,
+                    ));
+                    unresolved_functions.push_fn(self.module_id, func_id, impl_method);
+                }
+            }
```

Comment:
This block looks like it was copy-pasted twice

Reply:
Most likely an artifact from the git merge conflict resolution I had to do. I will review and fix this tomorrow...

Reply:
Fixed: https://github.com/noir-lang/noir/pull/2844/commits/bee0f58263ffb99609048b65d41c901786053e8f

It was indeed introduced by accident, during git merge conflict resolution.